### ISSUE TITLE
AI name assertion

### ISSUE THREAD

**Issue by robieta** [chikinn/hanabi#28]
During the last cleanup the encoding AI name check was changed from:

if i.split('_')[0] != 'Genencode':

to:

if i[:-1] != 'Gencoder':

The rationale for the original form is that it makes a less rigid assumption about the name string structure. For instance in the (admittedly extreme) case of a 10 player game, for the 10th player there would be a check on 'Gencoder1' == 'Gencoder'. More generally, the split on underscore can be used to have permutations on an AI for heterogeneous AI strategies. i.e. Example_StratA and Example_StratB would both pass an 'example' check. That said, I do not set the overall project style, so if you wish to simply enforce that all checks have the [:-1] == 'stuff': form for consistency I will comply going forward.

**Comment by fpvandoorn** [comment]
I agree that the new check is not very good. I've recently added the variable `PlayerRecord` to the `Round` class, so now we could check whether all the players are the same using something like this (which is completely independent to the names generate):

```
for p in r.PlayerRecord:
    if p.__class__.__name__ != 'HatPlayer':
        raise NameError('Hat AI must only play with other hatters')
```

**Comment by chikinn** [comment]
Looks good Floris!  I don't care how we check, I just want the player names themselves to be short and consistent (e.g., `encoder` and `gencoder` are respectively consistent with `cheater` and `brainbow`).

Sorry for stepping on toes.  Any check is fine with me and I like the idea of standardizing it in `Round`.
